lib/ansible/plugins/vars/host_group_vars.py

```diff
@@ -55,60 +55,92 @@
 
 import os
 from ansible.errors import AnsibleParserError
-from ansible.module_utils._text import to_bytes, to_native, to_text
+from ansible.module_utils._text import to_bytes, to_native
 from ansible.plugins.vars import BaseVarsPlugin
-from ansible.inventory.host import Host
-from ansible.inventory.group import Group
+from ansible.utils.path import basedir
 from ansible.utils.vars import combine_vars
 
+CANONICAL_PATHS = {}
 FOUND = {}  # type: dict[str, list[str]]
+NAK = set()
+HOSTS = set()
+GROUPS = set()
```

Comment:
the thing about adding the cache, is_stateless is now a lie, but used to reuse the cache ... should we rename the property?

Reply:
The caches are globals though, not tied to the instance.

Reply:
I renamed it to `cache_instance`, but feel free to bikeshed.